#! /bin/sh -e

# All lines beginning with `# DP:' are a description of the patch.
# DP: Description: Fix mips/mipsel compilation breakage with the recent binutils.
# DP: Related bugs: #262646: glibc_2.3.2.ds1-14_mips: FTBFS: {standard input}:162: Error: junk at end of line
# DP: Dpatch author: GOTO Masanori <gotom@debian.org>
#		     Thiemo Seufer <ica2_ts@csv.ica.uni-stuttgart.de>
# DP: Patch author: 
# DP: Upstream status: In CVS
# DP: Status Details: 
# DP: Date: 2004-08-02

PATCHLEVEL=0

if [ $# -ne 2 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch -d "$2" -f --no-backup-if-mismatch -p$PATCHLEVEL < $0;;
    -unpatch) patch -d "$2" -f --no-backup-if-mismatch -R -p$PATCHLEVEL < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

# append the patch here and adjust the -p? flag in the patch calls.
cvs -r 1.69 -r 1.71:

2004-04-15  Atsushi Nemoto <anemo@mba.ocn.ne.jp>

	* sysdeps/mips/dl-machine.h (RTLD_START): Do not use nested .end.


--- sysdeps/mips/dl-machine.h.orig	2004-08-01 21:02:05.000000000 +0200
+++ sysdeps/mips/dl-machine.h	2004-08-01 21:03:43.000000000 +0200
@@ -475,7 +475,7 @@ _dl_runtime_resolve:\n							      \
 	" STRINGXP(PTR_LA) " $25, _dl_start_user\n\
 	.globl _dl_start_user\n\
 	.type _dl_start_user,@function\n\
-	.ent _dl_start_user\n\
+	.aent _dl_start_user\n\
 _dl_start_user:\n\
 	" STRINGXP(SETUP_GP) "\n\
 	" STRINGXV(SETUP_GP64($18,_dl_start_user)) "\n\
@@ -513,8 +513,7 @@ _dl_start_user:\n\
 	" STRINGXP(PTR_LA) " $2, _dl_fini\n\
 	# Jump to the user entry point.\n\
 	move $25, $17\n\
-	jr $25\n\
-	.end _dl_start_user\n\t"\
+	jr $25\n\t"\
 	_RTLD_EPILOGUE(ENTRY_POINT)\
 	".previous"\
 );


